daemon/main.c

```diff
@@ -905,14 +912,33 @@ int main(int argc, char **argv) {
                             fprintf(stderr, "\n\nALL TESTS PASSED\n\n");
                             return 0;
                         }
+#ifdef ENABLE_DBENGINE
                         else if(strncmp(optarg, createdataset_string, strlen(createdataset_string)) == 0) {
                             optarg += strlen(createdataset_string);
-#ifdef ENABLE_DBENGINE
-                            unsigned history_seconds = (unsigned )strtoull(optarg, NULL, 0);
+                            unsigned history_seconds = strtoul(optarg, NULL, 0);
```

Comment:
`unsigned` type for the majority of the compilers is `unsigned integer`, but the function ``` strtoul``` returns `unsigned long' that is bigger than `unsigned` for 64 bits host, so I think it is better to keep the cast here.

Reply:
Done.